### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by danielgtaylor**
Initial commit of the refract element unique ID generation library. The test fixtures include lots of examples of what the generated IDs look like - let me know if that is okay!

cc @Baggz

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  "author": "Apiary, Inc.",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/apiaryio/abagnale.git"
+  },
+  "license": "MIT",
+  "devDependencies": {
+    "glob": "^7.0.0",
+    "peasant": "^0.5.2"
```
Comment (abtris) [inline package.json]: @danielgtaylor do you introduce this your library to engineering? Looks very good!

File: README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+A library to forge IDs for elements in [Refract](https://github.com/refractproject/refract-spec#refract) data structures. Data structures can come from [MSON](https://github.com/apiaryio/mson#markdown-syntax-for-object-notation) or other input sources.
+
+Abagnale attempts to give all elements within the structure a unique ID, even for elements it does not know. It accomplishes this by crawling the element contents and looking for structures that look like other elements.
+
+See the [test output](https://github.com/apiaryio/abagnale/tree/master/test/output) directory for examples of generated IDs.
+
+It is named after [Frank Abagnale](https://en.wikipedia.org/wiki/Frank_Abagnale), one of the most notorious tricksters ever known. He forged several fake IDs, a pilots license with which he flew over a million miles, faked being a college professor, worked as a fake chief resident pediatrician, and worked in the Louisiana State Attorney General's office with a fake degree from Harvard before being caught.
```
Comment (Baggz) [inline README.md]: Haha! :smile:

File: test/output/simple-types.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    "content": "test",
+    "meta": {
+      "id": "string"
+    }
+  },
+  {
+    "element": "string",
+    "content": "another test",
+    "meta": {
+      "id": "string2"
```
Comment (Baggz) [inline test/output/simple-types.json]: Not sure if we shouldn't put a separator between ‘string’ and ‘2’ (e.g. `string.2`, `string-2`).

File: test/output/simple-types.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    "content": 123,
+    "meta": {
+      "id": "number"
+    }
+  },
+  {
+    "element": "number",
+    "content": 456,
+    "meta": {
+      "id": "number2"
```
Comment (Baggz) [inline test/output/simple-types.json]: Same as https://github.com/apiaryio/abagnale/pull/1/files?short_path=04c6e90#r54010944.

**Comment by Baggz** [comment]
Looks good to me, @danielgtaylor! Let's release it and test it out!

File: package.json (hunk 1 above)
Comment (danielgtaylor) [inline package.json]: The API Blueprint team knows about it, but maybe not all of engineering. I can present it but at the moment it's quite basic and doesn't work with e.g. React. It also now uses an outdated Babel, so one of these days soon I should update Peasant.

File: test/output/simple-types.json (hunk 3 above)
Comment (danielgtaylor) [inline test/output/simple-types.json]: Good idea, updating this!

**Pull request merged**